File: 프로그래머스/0/181893. 배열 조각하기/배열 조각하기.kt
class Solution {
    fun solution(arr: IntArray, query: IntArray): IntArray {
        var result = arr.toList()
        for (i in query.indices) {
            result = if (i % 2 == 0) {
                result.subList(0, query[i] + 1)
            } else {
                result.subList(query[i], result.size)
            }
        }
        return result.toIntArray()
    }
}

File: 프로그래머스/0/181893. 배열 조각하기/배열 조각하기.js
function solution(arr, query) {
    for (let i = 0; i < query.length; i++) {
        if (i % 2 === 0) {
            // 짝수 인덱스: query[i] 이후는 잘라버림 (query[i] 포함)
            arr = arr.slice(0, query[i] + 1);
        } else {
            // 홀수 인덱스: query[i] 이전은 잘라버림 (query[i] 포함)
            arr = arr.slice(query[i]);
        }
    }
    return arr;
}
